gigascience/gigadb-website
# Dataset Sample Multiselect

# Pull request for issue: #2120 

This is a pull request for the following functionalities:

* Add a multiselect box and update the controller to handle multiple sample selections
![image](https://github.com/user-attachments/assets/e27c79f4-bf26-42c6-ad0a-859c625b18ea)

## How to test?

* navigate to http://gigadb.gigasciencejournal.com/adminDatasetSample/create
* select multiple samples and click on "create". To select multiple samples you can hold ctrl / cmd and click on multiple
* Verify the samples have been added in http://gigadb.gigasciencejournal.com/adminDatasetSample/admin

## How have functionalities been implemented?

- Replace the single select with a multiselect box
- Update the controller to create multiple DatasetSamples as a transaction
- If the method errors out, cancel the transaction and display an error in the create page as a "flash"

## Any issues with implementation?

the current behavior is the native browser behavior, which from the behaviors listed here (https://www.w3.org/WAI/ARIA/apg/patterns/listbox/) it is not the "recommended" one but the "alternative". Taking that into account I have tried to improve the widget in terms of a11y but the amount of work involved is significant for what I understand to be a rare use case, so left the native browser behavior unchanged

## Any changes to automated tests?

Added one test to verify that selecting multiple samples at once and creating the relation works
